src/kount/inquiry.py

```diff
@@ -187,7 +188,16 @@ def set_ip_address(self, ip_adr):
         """Set the IP address. ipaddress
         Arg: ip_adr - IP Address of the client
         """
-        self.set_param("IPAD", str(ip_adr))
+        ip = None
+        for cls in (ipaddress.IPv4Address, ipaddress.IPv6Address):
+            try:
+                ip = cls(ip_adr)
+                break
+            except ipaddress.AddressValueError:
+                pass
+        if not ip:
+            raise ValueError('invalid ip address %s' % ip_adr)
+        self.set_param("IPAD", ip_adr)
```

Comment:
I think this code can be simplified and made more readable using ipaddress.ip_address to determine if we are given a valid ip address; we can also remove the for loop because ipaddress.ip_address will check both IPv4 and IPv6. The error thrown if it is not valid also seems sufficient that we don't need to catch and rethrow it. Let me know if this helps!

Example errors:
ValueError: '192.168.0.15x' does not appear to be an IPv4 or IPv6 address
ValueError: '2001:0:3238:DFE1:63::FEasdasadFB' does not appear to be an IPv4 or IPv6 address

https://docs.python.org/3.4/library/ipaddress.html#convenience-factory-functions

```
def set_ip_address(self, ip_adr):
	"""Set the IP address. ipaddress
	Arg: ip_adr - IP Address of the client
	"""
	# Verify ip_adr is a valid IPv4 or IPv6 address
	# ipaddress.ip_address will throw a Value Error if not 
	ip = ipaddress.ip_address(ip_adr)
	self.set_param("IPAD", ip.exploded)
```

Reply:
Yes, definitely we should go this way, I've missed this method.  As long as the error thrown, I've made it throw ValueError as the other methods, but since ipaddress.AddressValueError is a subclass of ValueError we can use it directly instead of catching it.

Reply:
I concur with Constantine.  I think that is clean and simple, but gets the job done.

Reply:
Am I supposed to commit the proposed change, or you are going to update the PR?

And I have one more question, ipaddress.ip_address(ip_adr).exploded may return IP string different than ip_adr parameter, for example:

ipaddress.ip_address(**'FE80::0202:B3FF:FE1E:8329'**).exploded
**'fe80:0000:0000:0000:0202:b3ff:fe1e:8329'**

So the client will send unknowingly different string, which may be OK as long as he doensn't expect to see the exact same IP string later in some other response from the server.

Reply:
Please commit the proposed change.

Let's not use .exploded since it is changing the string. I think returning the ip_adr parameter is the way to go for consistency; good catch!